docker/dev_env/Dockerfile

```diff
@@ -1,21 +1,24 @@
-FROM docker.io/library/fedora:latest
+FROM docker.io/library/fedora:latest AS base-system
```

Comment:
Are you considering multi-stage builds for the dev-env image?

Reply:
I was, because I was getting frustrated with the `ENV` declarations causing the `dnf` runs to not be cached, but I'll remove this.

One thought I had was if the differences between macos and Linux hosts continue to be an issue, maybe having separate targets for each would help.